### DISCUSSION TITLE
Link prediction task and graph neural networks

### DISCUSSION THREAD

**Discussion by BarclayII** [snap-stanford/ogb#237] [WikiKG90M-LSC]
Hi all,

Big thanks for organizing the Large Scale Challenge!  It really showed interesting insights on deep learning on graphs, especially for link prediction, because it seems that the top-ranking models (which already got a near-perfect score of 0.97+ MRR) are all some sort of variants of knowledge graph embedding approaches instead of GNNs.  This is exactly the opposite to node classification and graph classification where the top solutions were all using GNNs.

I'm wondering if anybody in the community has ever tried a GNN-based approach on WikiKG90M.  Actually, the same question goes for other knowledge graph datasets on the OGB leaderboard (WikiKG2, BioKG).

**Comment by weihua916** [comment]
GNN for KG would be interesting to explore! I have not seen it in the leaderboard, but one related idea is to use GNN as the encoder (R-GCN paper, https://arxiv.org/abs/1703.06103).

On a side note, getting a near-perfect score of 0.97 MRR at WikiKG90M does *not* indicate that the problem is solved. Our task only asks to rank the correct tails among 1000 candidate entities, but **in practice, the model needs to rank against all the 90M entities**, which is far more challenging. Also, all the winning solutions exploit the candidate tail entity distributions, which boosts the performance a bit, but is not practical when considering ranking against all the 90M entities. We have asked the winners to clarify this limitation in their updated report.

In short, the WikiKG90M-LSC task can be made harder and more practically relevant. There is a lot to be done in the space. We will keep the community updated if we are to refresh the task.

**Reply by weihua916** [reply]
Another related paper is [SEAL](https://papers.nips.cc/paper/2018/file/53f0d7c537d99b3824f0f99d62ea2428-Paper.pdf). It may require some efforts to scale it to LSC graphs.

**Reply by weihua916** [reply]
[Neural Bellman-Ford Networks](https://arxiv.org/abs/2106.06935) and [ID-GNN](https://arxiv.org/abs/2101.10320) are providing a cool framework to recast the link prediction problem as a conditional node classification problem.
